docs/runtimes/image-build.md

```diff
@@ -0,0 +1,192 @@
+(build-function-image)=
+# Build function image
+
+As discussed in {ref}`images-usage`, MLRun provides pre-built images which contain the components necessary to execute
+an MLRun runtime. In some cases, however, custom images need to be created. 
+This page details this process and the available options.
+
+## When is a build required?
+
+In many cases an MLRun runtime can be executed without having to build an image. This will be true when
+the basic MLRun images fulfill all the requirements for the code to execute. It will be required to build an image 
+if one of the following is true:
+
+1. Code is using additional Python packages, OS packages, scripts or other configurations that need to be applied
+2. Using different base-images or different versions of MLRun images than provided by default
+3. Executed source code has changed, and the image has the code packaged in it - see
+   [here](mlrun_jobs.html#deploy-build-the-function-container) for more details on source code, and using 
+   {py:func}`~mlrun.runtimes.BaseRuntime.with_code()` to avoid re-building the image when the code has changed
+4. Running nuclio functions, which are packaged as images (build is triggered by MLRun and performed by nuclio)
```

Comment:
- The code runs nuclio functions, which are packaged as images (the build is triggered by MLRun and executed by nuclio).